src/main/java/com/patchworkmc/Patchwork.java

```diff
@@ -213,44 +221,66 @@ public static void transformMod(Path currentPath, Path jarPath, Path outputRoot,
 
 		LOGGER.trace("fabric.mod.json: " + json);
 
+		// Write patchwork logo
+
+		if (primary.getAsJsonPrimitive("icon").getAsString().equals("assets/patchwork-generated/icon.png")) {
+			Files.createDirectories(fs.getPath("assets/patchwork-generated/"));
+			Files.write(fs.getPath("assets/patchwork-generated/icon.png"), patchworkIcon);
+		}
+
 		try {
 			Files.createDirectory(fs.getPath("/META-INF/jars/"));
 		} catch (IOException ignored) {
 			// ignored
 		}
 
+		// If you touch this, you better be SURE it works properly before pushing.
+		// Yes, it's a hack, but it works.
+		// I spent several hours trying to make a cleaner solution to no avail.
+		// If you *really* want to clean this up, do *not* try to use ZipOutputStream
+		// - Glitch
 		for (JsonObject entry : mods) {
 			String modid = entry.getAsJsonPrimitive("id").getAsString();
-
 			if (entry == primary) {
 				// Don't write the primary jar as a jar-in-jar!
 				continue;
 			}
 
-			ByteArrayOutputStream jar = new ByteArrayOutputStream();
-			ZipOutputStream zip = new ZipOutputStream(jar);
+			// generate the jar
+			Path subJarPath = Paths.get("temp/" + modid + ".jar");
+			Map<String, String> env = new HashMap<>();
+			env.put("create", "true");
+			FileSystem subFs = FileSystems.newFileSystem(new URI("jar:" + subJarPath.toUri().toString()), env);
```

Comment:
Would it be possible to create a nested zipfs?

Reply:
I tried that in my original implementation a long time ago and it didn't work